gobblin-api/src/main/java/gobblin/configuration/State.java

```diff
@@ -98,6 +102,19 @@ public void setProp(String key, Object value) {
   }
 
   /**
+   * Appends the input value to a list property that can be retrieved with {@link #getPropAsList}.
+   * @param key property key
+   * @param value property value (should not include commas, which are used as separators)
```

Comment:
If you currently have `kafka.brokers=broker1,broker2`, you can certainly pass `broker3,broker4,broker5` as value to this method, so you know have 5 brokers, right?

It is true that if an individual broker's value has comma, then it will be problematic. However, the fact that comma is used as separator not only affects this method but also for other methods such as `setProp` and `getPropAsList`. So this should be explained in the Javadoc of this class, rather than here.